src/Sulu/Bundle/MediaBundle/Tests/Features/media_upload.feature

```diff
@@ -0,0 +1,32 @@
+Feature: Media upload
+    In order to organize my medias
+    As a user
+    I need to be able to upload medias
+
+    Background:
+        Given I am logged in as an administrator
+
+    Scenario: Upload new media
+        Given the media collection "Foobar" exists
+        When I am editing the media collection "Foobar"
+        And I attach the file "images/image1.png" to the current drop-zone
+        And I attach the file "images/image2.png" to the current drop-zone
+        And I attach the file "images/image3.jpg" to the current drop-zone
+        And I attach the file "images/image4.jpg" to the current drop-zone
+        Then I expect a success notification to appear
+        And I expect to see "4" ".datagrid-container .item" elements
+
+    Scenario: Upload new version
+         Given the media collection "Foobar" exists
+         And the file "image1.png" has been uploaded to the "Foobar" collection
+         When I am editing the media collection "Foobar"
+         Then I expect to see "1" ".item" elements
+         And I double click the data grid item "image1.png"
+         And I expect an overlay to appear
+         And I expect to see "Click or drag and drop new version"
+         And I attach the file "images/image2.png" to the current drop-zone
+         And I expect a success notification to appear
+         And I click the overlay tab "History"
+         And I expect to see "1" ".media-edit-versions-link" elements
```

Comment:
This series of `And` sentances breaks the semantics of the scenario.

`When` should be when you upload a new file, and `Then` should be the success notification.

Try:

````cucumber
Given the media collection...
And the file ...
And I wait to see "1" ".item" elements
And I wait for an overlay to appear
And I wait to see "Click or ...
When I attach the file ...
Then I expect a scucess notification
And ..
And ..
````

i.e. we change `I expect` to `I wait to`. You will need to add the alternative syntax for the `overlay` method in `AdminBundle\Behat\AdminContext`